### DISCUSSION TITLE
Add implicit sync support

### DISCUSSION THREAD

**Discussion by orowith2os** [NVIDIA/open-gpu-kernel-modules#532] [Ideas]
Linux is currently making a migration to Wayland, and in general better APIs like explicit sync, however NVIDIA is a major blocker here. 

NVIDIA absolutely refuses to support implicit sync, effectively pushing explicit sync in the Linux ecosystem before it's mature enough, and before the relevant software supports it. This breaks the user experience for *everyone* that isn't using the decades-old legacy display server (Xorg), and hinders progress.

NVIDIA should add support for implicit sync to their driver such that Xwayland functions well enough for most relevant use cases. Then progress can be made towards making Xwayland work better on an explicitly synced driver.

Everyone else that's not NVIDIA, even Nouveau, supports implicit sync. But the official NVIDIA drivers are the exception to the norm. You actively ship and build your drivers for Linux, so support it. Don't hold it back. 

You've already done some work on explicit sync support in [Xwayland](https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/967), and an alternative was brought up:

> Anyway, as an alternative to adding explicit sync at the protocol level, I was thinking another option would be to continue to use dma-buf implicit fences as the "IPC" mechanism, but have Xwayland use explicit sync internally.

Until then, NVIDIA should support implicit sync, or contribute more to Nouveau.

**Comment by gilvbp** [comment]
This problem has been discussed ( https://gitlab.freedesktop.org/xorg/xserver/-/merge_requests/1120 and here https://gitlab.freedesktop.org/xorg/xserver/-/issues/1317) for more than one year.
